tests/pykafka/test_balancedconsumer.py

```diff
@@ -78,28 +77,31 @@ def test_decide_partitions(self):
                 idx = participants.index(cns._consumer_id)
                 parts_per_consumer = num_partitions / num_participants
                 parts_per_consumer = math.floor(parts_per_consumer)
+
                 num_parts = parts_per_consumer + (0 if (idx + 1 > remainder_ppc) else 1)
 
-                self.assertEqual(len(partitions), num_parts)
+                self.assertEqual(len(partitions), int(num_parts))
 
             # Validate all partitions were assigned once and only once
             all_partitions = topic.partitions.values()
-            all_partitions.sort()
-            assigned_parts.sort()
+            all_partitions = sorted(all_partitions, key=lambda x: x.id)
+            assigned_parts = sorted(assigned_parts, key=lambda x: x.id)
             self.assertListEqual(assigned_parts, all_partitions)
 
 
-class BalancedConsumerIntegrationTests(unittest2.TestCase):
+class BalancedConsumerIntegrationTests(unittest.TestCase):
     maxDiff = None
 
     @classmethod
     def setUpClass(cls):
         cls.kafka = get_cluster()
-        cls.topic_name = 'test-data'
+        cls.topic_name = b'test-data'
```

Comment:
Right, so re "topic names as unicode": these tests should probably be required to work without this `b` here.